src/tools/imgtool/main.cpp

```diff
@@ -742,77 +739,64 @@ static int cmd_listfilters(const struct command *c, int argc, char *argv[])
 	return 0;
 }
 
-static void listoptions(const option_guide *opt_guide, const char *opt_spec)
+static void listoptions(const util::option_guide &opt_guide, const char *opt_spec)
 {
-	char opt_name[32];
-	const char *opt_desc;
-	util::option_resolution::range range[32];
-	char range_buffer[512];
-	char buf[32];
-	int i;
-
-	assert(opt_guide);
+	util::option_resolution resolution(opt_guide);
+	resolution.set_specification(opt_spec);
 
 	fprintf(stdout, "Option           Allowed values                 Description\n");
 	fprintf(stdout, "---------------- ------------------------------ -----------\n");
 
-	while(opt_guide->option_type != OPTIONTYPE_END)
+
+	for (auto iter = resolution.entries_begin(); iter != resolution.entries_end(); iter++)
 	{
-		range_buffer[0] = 0;
-		snprintf(opt_name, ARRAY_LENGTH(opt_name), "--%s", opt_guide->identifier);
-		opt_desc = opt_guide->display_name;
+		const auto &entry = *iter;
```

Comment:
Please don't use auto type for no good reason like this - it just obfuscates what you're trying to do.  It's not like entry is a proxy or iterator, it can be a reference to exactly one thing and you know exactly what it is.